### PR TITLE
[opt](routine-load) end Kafka consume when meets partition EOF

### PR DESCRIPTION
**Pull request by sollhui**
## Proposed changes

When all the partitions held by a consumer reach EOF, the consumer can end and give up resources to avoid ineffective occupation of resources.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HHoflittlefish777** [comment]
run buildall

**Comment by HHoflittlefish777** [comment]
run buildall

File: be/src/runtime/routine_load/data_consumer.h [hunk 1]
```diff
@@ -160,6 +160,7 @@ class KafkaDataConsumer : public DataConsumer {
     std::string _brokers;
     std::string _topic;
     std::unordered_map<std::string, std::string> _custom_properties;
+    std::set<int32_t> _eof_partition_ids;
```
Comment (liaoxin01) [inline be/src/runtime/routine_load/data_consumer.h:163]: _eof_partition_ids -> _consuming_partition_ids

**Comment by HHoflittlefish777** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**